applications/ConvectionDiffusionApplication/tests/bfecc_convection_test.py

```diff
@@ -0,0 +1,133 @@
+            for node in self.main_model_part.Nodes:
+                x_local = node.X - 0.5
+                y_local = node.Y - 0.5
+                r = math.sqrt(x_local**2 + y_local**2)
+                if(r < 0.45):
+                    node.SetSolutionStepValue(KratosMultiphysics.VELOCITY, 0, [-y_local, x_local, 0.0])
+                    node.SetSolutionStepValue(KratosMultiphysics.TEMPERATURE, 0, math.sqrt((x_local - x_c)**2 + (y_local - y_c)**2) - 0.1)
+
+            # Create the search structure
+            locator = KratosMultiphysics.BinBasedFastPointLocator2D(self.main_model_part)
+            locator.UpdateSearchDatabase()
+
+            # Construct the BFECC utility
+            bfecc_utility = ConvectionDiffusionApplication.BFECCConvection2D(locator)
+            self.main_model_part.CloneTimeStep(0.0)
+
+            # Set output
+            if (self.print_output):
+                gid_mode = KratosMultiphysics.GiDPostMode.GiD_PostBinary
+                multifile = KratosMultiphysics.MultiFileFlag.SingleFile
+                deformed_mesh_flag = KratosMultiphysics.WriteDeformedMeshFlag.WriteUndeformed
+                write_conditions = KratosMultiphysics.WriteConditionsFlag.WriteElementsOnly
+                gid_io = KratosMultiphysics.GidIO(self.reference_file, gid_mode, multifile, deformed_mesh_flag, write_conditions)
+
+                mesh_name = 0.0
+                gid_io.InitializeMesh(mesh_name)
+                gid_io.WriteMesh(self.main_model_part.GetMesh())
+                gid_io.FinalizeMesh()
+                gid_io.InitializeResults(mesh_name,(self.main_model_part).GetMesh())
+
+            # Advance in time and convect
+            t = 0.0
+            while(t < self.end_time):
+                t += self.dt
+                self.main_model_part.CloneTimeStep(t)
+
+                bfecc_utility.BFECCconvect(
+                    self.main_model_part,
+                    KratosMultiphysics.TEMPERATURE,
+                    KratosMultiphysics.VELOCITY,
+                    self.bfecc_substeps)
+
+                if (self.print_output):
+                    gid_io.WriteNodalResults(KratosMultiphysics.VELOCITY, self.main_model_part.Nodes, t, 0)
+                    gid_io.WriteNodalResults(KratosMultiphysics.TEMPERATURE, self.main_model_part.Nodes, t, 0)
+
+            if (self.print_output):
+                gid_io.FinalizeResults()
+
+    def checkResults(self):
+        with UnitTest.WorkFolderScope(self.work_folder, __file__):
+            if self.print_reference_values:
+                with open(self.reference_file + '.csv','w') as reference_file:
+                    reference_file.write("#ID, TEMPERATURE\n")
+                    for node in self.main_model_part.Nodes:
+                        reference_file.write("{0}, {1}\n".format(node.Id, node.GetSolutionStepValue(KratosMultiphysics.TEMPERATURE, 0)))
+            else:
+                with open(self.reference_file + '.csv','r') as reference_file:
```

Comment:
why manually?
I recommend using the check processes from the core

Reply:
Done.